### PR TITLE
Halit week 6

### PR DESCRIPTION
**Pull request by halitos**
# Your Details

Your Name: Halit Oskan
Your City: Manchester
Your Slack Name: Halit

# Homework Details

Module: JS 2-3 
Week: 6

### CODE REVIEW

File: week-6/Homework/mandatory/1-alarmclock/alarmclock.js [hunk 1]
```diff
@@ -1,7 +1,32 @@
...
+    if (inputVal >= 0) {
+      document.getElementById(
+        "timeRemaining"
+      ).innerText = `Time Remaining: ${sec2time(inputVal)} `;
+      inputVal -= 1;
+      if (inputVal < 0) {
+        playAlarm();
+      }
+    }
+  }
```
Comment (Ashaghel) [inline week-6/Homework/mandatory/1-alarmclock/alarmclock.js:20]: the code is really good
the only issue is readability 
no need to declare a function inside another function 
DOM elements declaration can be outside 
names of functions, sec2time ?

File: week-6/Homework/mandatory/2-quotegenerator/quotes.js [hunk 2]
```diff
@@ -489,3 +491,33 @@ const quotes = [
...
   },
 ];
+
+function pickFromArray(choices) {
+  return choices[Math.floor(Math.random() * choices.length)];
+}
+
+const containerDiv = document.createElement("div");
+containerDiv.style.width = "800px";
+containerDiv.style.height = "100%";
```
Comment (Ashaghel) [inline week-6/Homework/mandatory/2-quotegenerator/quotes.js:501]: do you need the height to be 100%?

File: week-6/Homework/mandatory/2-quotegenerator/quotes.js [hunk 3]
```diff
@@ -489,3 +491,33 @@ const quotes = [
...
+
+const quoteLine = document.createElement("h1");
+const authorLine = document.createElement("h3");
+
+const nextButton = document.createElement("button");
+nextButton.innerText = "Next Quote";
+
+function loadQuote() {
+  const chosenQuote = pickFromArray(quotes);
+  quoteLine.innerText = '"' + chosenQuote.quote + '"';
```
Comment (Ashaghel) [inline week-6/Homework/mandatory/2-quotegenerator/quotes.js:512]: very understandable code here 🥇

File: week-6/Homework/mandatory/2-quotegenerator/style.css [hunk 4]
```diff
@@ -1 +1,23 @@
...
+  padding: 20px;
+}
+
+h3 {
+  padding: 30px;
+}
+
+button {
+  background-color: aqua;
+  margin-top: 20px;
```
Comment (Ashaghel) [inline week-6/Homework/mandatory/2-quotegenerator/style.css:22]: styling looks a bit weird as the UX feels tiring trying to find the button after each click

File: week-6/Homework/mandatory/3-slideshow/slideshow.js [hunk 5]
```diff
@@ -1 +1,75 @@
...
+function moveForward() {
+  imgContainer.setAttribute("src", `${slideImages[i]}`);
+  i++;
+  j = i - 2;
+  if (i === slideImages.length) {
+    i = 0;
+  } else {
+    i = i;
+  }
+}
```
Comment (Ashaghel) [inline week-6/Homework/mandatory/3-slideshow/slideshow.js:21]: interesting approach with two variables to control the position

File: week-6/Homework/mandatory/3-slideshow/slideshow.js [hunk 6]
```diff
@@ -1 +1,75 @@
...
+const backBtn = document.createElement("button");
+backBtn.innerText = "Back";
+document.body.appendChild(backBtn);
+backBtn.addEventListener("click", moveBack);
+
+autoBackBtn.addEventListener("click", function () {
+  var back = setInterval(moveBack, 2000);
+  stopBtn.addEventListener("click", function () {
+    clearInterval(back);
+  });
```
Comment (Ashaghel) [inline week-6/Homework/mandatory/3-slideshow/slideshow.js:48]: before setting the intervals you need to clear them, as clicking autoforward, autobackward, set you in a loop between two images

File: week-6/Homework/mandatory/3-slideshow/slideshow.js [hunk 7]
```diff
@@ -1 +1,75 @@
...
+nextBtn.addEventListener("click", moveForward);
+
+// auto forward button
+
+const autoForwardBtn = document.createElement("button");
+autoForwardBtn.innerText = "Auto-Forward";
+document.body.appendChild(autoForwardBtn);
+
+autoForwardBtn.addEventListener("click", function () {
+  var myVar = setInterval(moveForward, 2000);
```
Comment (Ashaghel) [inline week-6/Homework/mandatory/3-slideshow/slideshow.js:71]: try to be consistent with naming variables and function, the previous setinterval is called back, this should be forward maybe, myVar is a bit weird

File: week-6/Homework/mandatory/2-quotegenerator/quotes.js (hunk 2 above)
Comment (halitos) [reply week-6/Homework/mandatory/2-quotegenerator/quotes.js]: No need, moved styling to CSS file.

File: week-6/Homework/mandatory/2-quotegenerator/style.css (hunk 4 above)
Comment (halitos) [reply week-6/Homework/mandatory/2-quotegenerator/style.css]: Modified style, fixed the next button's position.

File: week-6/Homework/mandatory/3-slideshow/slideshow.js (hunk 6 above)
Comment (halitos) [reply week-6/Homework/mandatory/3-slideshow/slideshow.js:51]: I think I cleared them now, but still some bugs need to sort. Like auto buttons try to run each time clicked before resetting the previous click.

File: week-6/Homework/mandatory/3-slideshow/slideshow.js (hunk 7 above)
Comment (halitos) [reply week-6/Homework/mandatory/3-slideshow/slideshow.js]: Agree, bad variable name. Changed it.

File: week-6/Homework/mandatory/1-alarmclock/alarmclock.js (hunk 1 above)
Comment (halitos) [reply week-6/Homework/mandatory/1-alarmclock/alarmclock.js:20]: - no need to declare a function inside another function
- DOM elements declaration can be outside
Tried but could not do these.

**Pull request closed without merging**